test/oauth2_test.dart

```diff
@@ -14,14 +15,14 @@ import 'package:http/http.dart';
 
 import 'test_utils.dart';
 
+final _defaultResponse = Response('', 500);
+final _defaultResponseHandler = (Request _) async => _defaultResponse;
+
 main() {
   test('access-token', () {
     var expiry = new DateTime.now().subtract(const Duration(seconds: 1));
     var expiryUtc = expiry.toUtc();
 
-    expect(() => new AccessToken(null, 'bar', expiryUtc), throwsArgumentError);
```

Comment:
Should we be looking at adding a test with opted out code to retain this behavior? We expect it to still throw an error in unsound mode right? I think it's fine if the type of the error changes, but we should be sure that a call like this doesn't have other negative side effects.

Reply:
I was planning on removing the runtime null checks that are now redundant in sound mode, but I don't feel strongly about that. Agreed if we keep them I should test them. Please let me know if you would prefer

 - remove the checks and the tests
 - keep the tests, move them to mixed mode tests; if so how would you suggest I do that please? e.g. new test file with a language version comment would do it...

Looking at what happens if these are accidentally null, either there is an exception later or authentication will fail later. 

Thanks.

Reply:
I think given that these are Error today and shouldn't be caught it shouldn't be a problem that they become exceptions instead. I don't think we need specific tests for this behavior.

In case it comes up in some other package - yes the way to write unsound tests is to create a separate test file (I have used the name `opt_out_test.dart`) and add the language version comment.